### ISSUE TITLE
Invalid integer part  Invalid digit , found

### ISSUE THREAD

**Issue by cmodijk** [moneyphp/money#593]
I'm using version `3.3.0` and i'm getting the following error.

```
Invalid integer part 25,9500. Invalid digit , found
```

Example code 
```php
$moneyParser = new IntlLocalizedDecimalParser(
    new \NumberFormatter(
        'nl_NL',
        \NumberFormatter::DECIMAL
    ),
    new ISOCurrencies()
);

$moneyParser->parse(
    '25,95',
    new Currency('EUR')
);
```

The PHP version i'm using is `7.3.8`

I also tried the example from the website with cents and it also gives me the same error

```
Invalid integer part 1000,100. Invalid digit , found
```

```diff
$currencies = new ISOCurrencies();

$numberFormatter = new \NumberFormatter('nl_NL', \NumberFormatter::DECIMAL);
$moneyParser = new IntlLocalizedDecimalParser($numberFormatter, $currencies);

--$money = $moneyParser->parse('1.000,00', new Currency('EUR'));
++$money = $moneyParser->parse('1.000,10', new Currency('EUR'));
```

**Comment by edanao** [comment]
got the same issue. 
php version: 7.4.6 

`
$money = new Money('3350', new Currency('DKK');
$currencies = new ISOCurrencies();
$numberFormatter = new \NumberFormatter('da_DK', \NumberFormatter::DECIMAL);
$moneyFormatter = new IntlMoneyFormatter($numberFormatter, $currencies);
 $moneyFormatter->format($money);
`

`
 "error": "InvalidArgumentException",
"message": "Invalid integer part 33,5. Invalid digit , found",
`

**Comment by frederikbosch** [comment]
The outcome on my machine is this, php 7.3.18.

```php
/srv/libraries/money/test.php:19:
class Money\Money#6 (2) {
  private $amount =>
  string(4) "2595"
  private $currency =>
  class Money\Currency#5 (1) {
    private $code =>
    string(3) "EUR"
  }
}
```

Does your machine have the right locales installed?

**Comment by cmodijk** [comment]
For other people that come on this issue I got the error again on a other project and after debugging I found the problem. Somewhere in our application the local was set like this `setlocale(LC_ALL, 'nl_NL');` this make al the `floats` in PHP act like the local version which for `nl_NL` is `25,95` instead of `25.95`. Money PHP then converts this into a string and then has the wrong notation for parsing this into the money object. I solved this by switching back to `setlocale(LC_ALL, 'en_US');` just before using the parsing. 

Example code 
```php
setlocale(LC_ALL, 'nl_NL');

$numberFormatter = new \NumberFormatter('nl_NL', \NumberFormatter::DECIMAL);
var_dump($numberFormatter->parse('25,95'));
var_dump((string) $numberFormatter->parse('25,95'));
```

This is the output for `7.2.0 - 7.2.33, 7.3.0 - 7.3.21, 7.4.0 - 7.4.9` in `8.0.0alpha1 - beta1` this seems to be solved.
```output
float(25,95)
string(5) "25,95"
```

Resources
- https://github.com/moneyphp/money/blob/master/src/Parser/IntlLocalizedDecimalParser.php#L63
- https://3v4l.org/Fdvq5
